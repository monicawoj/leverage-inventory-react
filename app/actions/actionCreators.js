// import constants

// export action creator functions that return action object with type and payload
