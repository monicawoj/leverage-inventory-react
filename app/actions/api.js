//dev
// const id = 4260ccb2-60fe-46ba-8122-7ccd233cbf2d
// const url = "https://leverage-inventory-dev.wharton-research-programming.org/backend/results/";

//production
// const userId = getCookie('resultsid');
// const url = "https://leverage-inventory.wharton-research-programming.org/backend/results/";

//local
//const finalUrl = 'https://app.leverageinventory.com/backend/results/?id=4260ccb2-60fe-46ba-8122-7ccd233cbf2d';

// export default finalUrl;

const getApiUrl = (id) => {
  const base = process.env.API_URL;
  // console.log(base);
  return `${base}?id=${id}`;
}

export default getApiUrl;
