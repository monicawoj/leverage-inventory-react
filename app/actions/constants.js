export const GET_USER = 'GET_USER';
export const CHANGE_VIEW = 'CHANGE_VIEW';
export const USER_LOADED = 'USER_LOADED';
export const USER_LOADING_ERROR = 'USER_LOADING_ERROR';
export const CHANGE_COMPARISON_GROUP = 'CHANGE_COMPARISON_GROUP';
