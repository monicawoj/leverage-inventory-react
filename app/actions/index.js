export {
  getUser,
  changeView,
  userLoaded,
  userLoadingError,
  changeComparisonGroup
} from './actionCreators'
