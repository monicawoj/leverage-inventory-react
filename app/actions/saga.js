import { call, put, select, takeLatest } from 'redux-saga/effects';
// import constants
// import api information
// import action creator functions from index

// worker saga that performs the task
export function* workerSaga(action) {

  // try {
  //
  // } catch (error) {
  //
  // }
}

// watcher saga that watches for an action and executes worker saga on said action
export default function* watcherSaga() {
  yield takeLatest(ACTION_NAME, workerSaga);
}
