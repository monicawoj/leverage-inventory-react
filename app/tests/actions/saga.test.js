import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_BEERS, CHOOSE_FOCUS_BEER } from 'actions/constants';
import {
  beersLoaded,
  beersLoadingError,
  similarBeersLoaded,
  similarBeersLoadingError
} from 'actions/index';

import beerData, { getBeers, getSimilarBeers } from 'actions/saga';

const beer = {
  id: 3,
  abv: 4,
  ibu: 10,
  ebc: 10
};

describe('getBeers Saga function', () => {
  let getBeersGenerator;

  beforeEach(() => {
    const action = {page: 1};
    getBeersGenerator = getBeers(action);

    const selectDescriptor = getBeersGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

  });

  it('should dispatch the beersLoaded action if it requests the data successfully', () => {
    const response = [{
      name: 'First beer',
    }, {
      name: 'Second beer',
    }];
    const putDescriptor = getBeersGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(beersLoaded(response)));
  });

  it('should call the beerLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getBeersGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(beersLoadingError(response)));
  });
});

describe('getSimilarBeers Saga function', () => {
  let getSimilarBeersGenerator;

  beforeEach(() => {
    const beer = {
      id: 3,
      abv: 4,
      ibu: 10,
      ebc: 10
    };
    getSimilarBeersGenerator = getSimilarBeers({beer});

    const selectDescriptor = getSimilarBeersGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

  });

  it('should dispatch the similarBeersLoaded action if it requests the data successfully', () => {
    const response = [{
      name: 'First beer',
    }, {
      name: 'Second beer',
    }];
    const putDescriptor = getSimilarBeersGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(similarBeersLoaded(response)));
  });

  it('should call the error action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getSimilarBeersGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(similarBeersLoadingError(response)));
  });
});

describe('beerDataSaga Saga', () => {
  const beerDataSaga = beerData();

  it('should start task to watch for LOAD_BEERS action', () => {
    const takeLatestDescriptor = beerDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_BEERS, getBeers));
  });

  it('should start task to watch for CHOOSE_FOCUS_BEER action', () => {
    const takeLatestDescriptor = beerDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(CHOOSE_FOCUS_BEER, getSimilarBeers));
  });
});
