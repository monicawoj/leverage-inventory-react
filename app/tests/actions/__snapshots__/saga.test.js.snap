// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBeers Saga function should call the beerLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.punkapi.com/v2/beers?per_page=20&page=1",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getBeers Saga function should dispatch the beersLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.punkapi.com/v2/beers?per_page=20&page=1",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getSimilarBeers Saga function should call the error action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.punkapi.com/v2/beers?per_page=4&ibu_lt=20&ibu_gt=0&abv_lt=7&abv_gt=1&ebc_lt=20&ebc_gt=4",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getSimilarBeers Saga function should dispatch the similarBeersLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.punkapi.com/v2/beers?per_page=4&ibu_lt=20&ibu_gt=0&abv_lt=7&abv_gt=1&ebc_lt=20&ebc_gt=4",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
